### ISSUE TITLE
Bad argument to 'to_utfindex'

### ISSUE THREAD

**Issue by AkisArou** [iguanacucumber/magazine.nvim#6]
### documentation & FAQ (at the bottom)

- [X] I have checked the [documentation & FAQ (at the bottom)](https://github.com/iguanacucumber/magazine.nvim/blob/main/doc/cmp.txt) and it didn't resolve my problem.

### Minimal reproducible full config

Basic config reproduces this

### Description

I think PR #5 introduced a bug!
This happens for each source.

```vim/_editor.lua:0: InsertEnter Autocommands for "*"..script nvim_exec2() called at InsertEnter Autocommands for "*":0../home/akisarou/.local/share/nvim/lazy/cmp-cmdline/after/plugin/cmp_cmdline.lua: Vim(source):E5113: Error while calling lua chunk: ...arou/.local/share/n
vim/lazy/nvim-cmp/lua/cmp/context.lua:56: bad argument #2 to 'to_utfindex' (number expected, got string)
stack traceback:
^I[C]: in function 'to_utfindex'
^I...arou/.local/share/nvim/lazy/nvim-cmp/lua/cmp/context.lua:56: in function 'new'
^I...arou/.local/share/nvim/lazy/nvim-cmp/lua/cmp/context.lua:25: in function 'empty'
^I...arou/.local/share/nvim/lazy/nvim-cmp/lua/cmp/context.lua:44: in function 'new'
^I...kisarou/.local/share/nvim/lazy/nvim-cmp/lua/cmp/core.lua:28: in function 'new'
^I...kisarou/.local/share/nvim/lazy/nvim-cmp/lua/cmp/init.lua:12: in main chunk
^I[C]: in function 'require'
^I...share/nvim/lazy/cmp-cmdline/after/plugin/cmp_cmdline.lua:1: in main chunk
^I[C]: in function 'nvim_exec2'
^Ivim/_editor.lua: in function 'cmd'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:506: in function <...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:505>
^I...
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:453: in function 'source_runtime'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:424: in function 'packadd'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:356: in function '_load'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:197: in function 'load'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:352: in function <...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:351>
^I[C]: in function 'xpcall'
^I.../.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/util.lua:135: in function 'try'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:351: in function '_load'
^I...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/loader.lua:197: in function 'load'
^I...hare/nvim/lazy/lazy.nvim/lua/lazy/core/handler/event.lua:85: in function <...hare/nvim/lazy/lazy.nvim/lua/lazy/core/handler/event.lua:72>

# stacktrace:
  - vim/_editor.lua:0 _in_ **cmd**
```

### Steps to reproduce

Let magazine.nvim load

### Expected behavior

Should not error

### Actual behavior

Crashes when loaded

### Additional context

NVIM v0.10.2
Build type: RelWithDebInfo
LuaJIT 2.1.1727870382
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions     -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security     -fstack-clash-protection -fcf-protection     -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/neovim/src=/usr/src/debug/neovim -flto=auto -O2 -g -Og -g -flto=auto -fno-fat-lto-objects -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fsigned-char -fstack-protector-strong -Wno-conversion -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=always  -DUNIT_TESTING -DHAVE_UNIBILIUM -D_GNU_SOURCE -DINCLUDE_GENERATED_DECLARATIONS -I/usr/include/luajit-2.1 -I/usr/include -I/build/neovim/src/neovim/build/src/nvim/auto -I/build/neovim/src/neovim/build/include -I/build/neovim/src/neovim/build/cmake.config -I/build/neovim/src/neovim/src

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

**Comment by astridlyre** [comment]
@AkisArou I am getting this issue too. Is this an issue with release NVIM vs Nightly perhaps? I'm on the same version.

https://github.com/neovim/neovim/blob/989a37a594649528f28432388c0e7e28e8be2753/runtime/lua/vim/_editor.lua#L816

Seems to show the function with the expected signature for [this](https://github.com/iguanacucumber/magazine.nvim/pull/5) PR, but when I checkout the code for v0.10.2 on my local, I can't find the same functions, so my guess is this is some breaking change?

**Comment by Landaman** [comment]
I have the same problem! v0.10.2

**Comment by svampkorg** [comment]
Can confirm this seems to be an issue with release v0.10.2.  Just installed nightly to check, and do not get this error anymore then.

**Comment by iguanacucumber** [comment]
i'm going to add some checks to only use the newer vim.str_utfindex on nightly, can't push rn will do it tonight

**Comment by 4rtemis-4rrow** [comment]
no, the issue is caused by the plugin itself

I have 2 configs, one of them is using version 0.2.1, branch main, commit d224bdb

the second one is using the latest

I'm getting this error only on the second config

**Comment by iguanacucumber** [comment]
> no, the issue is caused by the plugin itself

i'm adapting the soon deprecated vim.str_utfindex function to it's newer implementation, i just forgot the newer implementation isn't there yet in stable releases

Anyway i fixed the issue 🎉 ([commit](https://github.com/iguanacucumber/magazine.nvim/commit/0faba72dc56baacf42361e23d88548ae1662f669) + i released version 0.4.1)
